### PR TITLE
[Async] Enable async/await and update UIExplorer and tests

### PR DESCRIPTION
**Pull request by ide**
- Enables async/await in .babelrc and transformer.js
- Adds regenerator to package.json. Users still need to explicitly require the regenerator runtime -- this is so that you only pay for what you use.
- Update AsyncStorage examples in UIExplorer to use async/await
- Update promise tests in UIExplorer to use async/await in addition to the promise API

Test Plan: AsyncStorage demo in UIExplorer works as expected. UIExplorer tests pass.

### CODE REVIEW

**Comment by spicyj** [comment]
Naive question: is there a way we can have regenerator required automatically but only when you use async/await?

**Comment by ide** [comment]
@spicyj sort of -- Babel's "runtime" option says that it pulls in babel-runtime/regenerator only when needed. The thing is "runtime" also pulls in core-js which polyfills Map, Set, and other ES6 features that RN already polyfills (or that JSC already provides if you're targeting iOS 8+).

**Comment by sahrens** [comment]
AsyncStorage example looks good to me - @frantic, can you import and make sure everything else works internally?

File: package.json [hunk 1]
```diff
@@ -45,36 +45,37 @@
     "react-native-start": "packager/packager.sh"
```
Comment (frantic) [inline package.json]: are changes in this file related?

Comment (ide) [inline package.json]: Yeah -- Babel has this hack where it patches ast-types and I needed to update the dependencies in RN's package.json to get things working.

**Pull request closed without merging**